internal/dockerservice/dockerservice.go

```diff
@@ -52,74 +52,93 @@ type Container struct {
 func NewContainer(name string, config container.Config, hostConfig container.HostConfig, networkConfig network.NetworkingConfig) Container {
 	return Container{
 		Name:          name,
-		Ctx:           context.Background(),
 		Config:        config,
 		HostConfig:    hostConfig,
 		NetworkConfig: networkConfig,
 	}
 }
 
-func (d Docker) GetContainer(ctx context.Context, name string) (Container, error) {
-	c := Container{}
-	containers, err := d.Cli.ContainerList(ctx, types.ContainerListOptions{})
+// GetContainer returns a docker container with the provided name (if any exists).
+// if no match exists, it returns a nil container and a nil error.
+func (d Docker) GetContainer(ctx context.Context, name string) (*Container, error) {
```

Comment:
not sure we need to return a Pointer here.

Reply:
yeah, I wanted it to be possible to return `nil` when no match is found. That way callers can handle such cases instead of inspecting the error.

Reply:
I see. That's a good use-case.